### ISSUE TITLE
Private torrent

### ISSUE THREAD

**Issue by skelder** [mono/monotorrent#118]
So after successfully downloading a public ubuntu distro torrent, I decided to test out the private torrent that our team is going to use, and I encountered some problem.

`System.InvalidOperationException` with message `You cannot add external peers to a private torrent` was being thrown for each peer. 
On line 296 inside the `Initialize` function in the `TorrentManager.cs`

Our current setup:

(Everything is being done in one local network, for simplicity)
1 raspberry pi 3 b+ running our own little private tracker(opentracker)
1 raspberry pi 3 b+ running Transmission torrent client
1 Chromebook running Transmission torrent client
1 Windows machine running Transmission torrent client

I create a private torrent with our tracker address added in the tracker list from the Transmission client, load that file on all computers. 
3 Transmission clients on 3 different OSes will happily share among themselves.(We usually start seeding from windows client, but that doesn't really matter i guess)

Now I load the same torrent file onto the `SampleClient` it gives me the above error and shuts down.

What I tried so far was to disable DHT and local peer discovery in the ClientEngine.cs file with the boolean flags. 
And I also tried commenting out the portions of the code that loaded the DHT file.

One strange thing is that whenever the error occurs I can see the list of peers and their IPs and they all match the 3 clients we are running for testing. which confuses me even further.

I think monotorrent does support the private torrent and private trackers, or am i wrong?

**Comment by alanmcgovern** [comment]
You're right, it should work.

This was a regression introduced during the [wide-sweeping change](https://github.com/mono/monotorrent/commit/ea3fbdbbf45d9534f30e384080515711958c5b48#diff-93c18760ce61cabfe80a5bffc7422f55L629) to move to an async/await codebase. Specifically this is where the check was accidentally applied to peers retrieved from Trackers, instead of just being applied to peers retrieved from DHT/PEX/LPD etc.


The change would be to split the AddPeersAsync method back into two parts so that when the Tracker adds peers they do not go through that particular `if` statement. If you're feeling like submitting a patch, one way to implement it would be to add a second overload of [AddPeerAsync and AddPeersAsync](https://github.com/mono/monotorrent/blob/master/src/MonoTorrent/MonoTorrent.Client/Managers/TorrentManager.cs#L615-L648) which looks like this:

```
public Task<bool> AddPeerAsync (Peer peer)
    => AddPeerAsync (peer, false);

internal async Task<bool> AddPeerAsync (Peer peer, bool fromTrackers)
{
    if (HasMetadata && Torrent.IsPrivate && !fromTrackers)
        throw new InvalidOperationException ("You cannot add external peers to a private torrent");
}

public Task<bool> AddPeersAsync (IEnumerable<Peer> peers)
    => AddPeersAsync (peers, false);

internal async Task<bool> AddPeersAsync (IEnumerable<Peer> peers, bool fromTrackers)
{
    // same checks as above...
}
```

Then the tracker would call: [await AddPeersAsync (e.Peers, true);](https://github.com/mono/monotorrent/blob/4f244d39f15d5c6a0cc79987ff2ba6a7c893322f/src/MonoTorrent/MonoTorrent.Client/Managers/TorrentManager.cs#L307).

I should be able to do that, add a test, and verify nothing else has gone wrong in this area later on today if you're still blocked by this.

**Comment by skelder** [comment]
Thanks for the suggestion, I will try to tinker around a bit and see if i can make my little set up working.

**Comment by skelder** [comment]
I applied your fix and it seems to have no issues. I also created a pull request with fix applied #119
